Core/src/RequestWrapper.php

```diff
@@ -282,38 +283,41 @@ private function applyHeaders(RequestInterface $request)
         ];
 
         if ($this->shouldSignRequest) {
-            $headers['Authorization'] = 'Bearer ' . $this->getToken();
-        }
+            $quotaProject = $this->quotaProject;
+            $token = null;
+
+            if ($this->accessToken) {
+                $token = $this->accessToken;
+            } else {
+                $credentialsFetcher = $this->getCredentialsFetcher();
+                $token = $this->fetchCredentials($credentialsFetcher);
+                $quotaProject = $credentialsFetcher->getQuotaProject() ?: $quotaProject;
+            }
 
-        return Psr7\modify_request($request, ['set_headers' => $headers]);
-    }
+            $headers['Authorization'] = 'Bearer ' . $token;
 
-    /**
-     * Gets the access token.
-     *
-     * @return string
-     */
-    private function getToken()
-    {
-        if ($this->accessToken) {
-            return $this->accessToken;
+            if ($this->quotaProject) {
```

Comment:
This should be `if ($quotaProject) {...}` I believe

Reply:
Nice find, thank you - updated!